src/registrar/templates/django/admin/includes/detail_table_fieldset.html

```diff
@@ -221,11 +221,99 @@ <h2 class="usa-modal__heading">
         </div>
         
         {% if original_object.action_needed_reason_email %}
-        <input id="last-sent-email-content" class="display-none" value="{{original_object.action_needed_reason_email}}">
+        <input id="last-sent-action-needed-email-content" class="display-none" value="{{original_object.action_needed_reason_email}}">
         {% else %}
-        <input id="last-sent-email-content" class="display-none" value="None">
+        <input id="last-sent-action-needed-email-content" class="display-none" value="None">
         {% endif %}
 
+    {% elif field.field.name == "rejection_reason_email" %}
+        <div class="margin-top-05 text-faded field-rejection_reason_email__placeholder">
+            &ndash;
+        </div>
+
+        {{ field.field }}
+
+        <button
+            aria-label="Edit email in textarea"
+            type="button"
+            class="usa-button usa-button--unstyled usa-button--dja-link-color usa-button__small-text margin-left-1 text-no-underline field-rejection_reason_email__edit flex-align-self-start"
```

Comment:
Since these buttons/ modals will never exist on the same page at the same time, I think there's an opportunity to significantly simplify the HTML and the JS by using generic handles (email-edit instead of field-rejection_reason_email__edit).

On the JS: You can now trim down config by moving a lot of the handles to the parent class

On the template: You can probably pull out the modals into an include that's a variant of modal.html, if not modal.html itself (that would be ideal)

It's not insignificant and your code has nothing wrong with it, so I won' block if you decide not to do it, but we should capture this plan for the next custom email ticket if it exists.